lib/publiccloud/azure.pm

```diff
@@ -72,18 +73,45 @@ sub find_img {
     my $container = $self->container;
     my $storage_account = get_var('PUBLIC_CLOUD_STORAGE_ACCOUNT', 'eisleqaopenqa');
     my $key = $self->get_storage_account_keys($storage_account);
-    my $cmd_show = "az storage blob show --account-key $key -o json " .
-      "--container-name '$container' --account-name '$storage_account' --name '$name' " .
-      '--query="{name: name,createTime: properties.creationTime,md5: properties.contentSettings.contentMd5}"';
-    record_info('BLOB INFO', script_output($cmd_show, proceed_on_failure => 1));
-
-    my $json = script_output("az image show --resource-group " . $self->resource_group . " --name $name", 60, proceed_on_failure => 1);
-    record_info('IMG INFO', $json);
-    my $image;
+
+    $json = script_output("az storage blob show --account-key $key -o json " .
+          "--container-name '$container' --account-name '$storage_account' --name '$name' " .
+          '--query="{name: name,createTime: properties.creationTime,md5: properties.contentSettings.contentMd5}"',
+        proceed_on_failure => 1);
+    record_info('BLOB INFO', $json);
     eval {
-        $image = decode_azure_json($json)->{name};
+        $md5 = decode_azure_json($json)->{md5};
     };
-    record_info('IMG NOT-FOUND', "Cannot find image $name. Need to upload it.\n$@") if ($@);
+    if ($@) {
+        record_info('BLOB NOT-FOUND', "Cannot find blob $name. Need to upload it.\n$@");
+    } elsif (!$md5 || $md5 !~ /^[a-fA-F0-9]{32}$/) {
+        record_info('INVALID', "The blob $name does not have valid md5 field.");
+        return $image;
+    }
+
+    my $arch = (check_var('PUBLIC_CLOUD_ARCH', 'arm64')) ? 'Arm64' : 'x64';
+    if ($arch eq 'Arm64') {
+        my $resource_group = $self->resource_group;
+        my $gallery = $self->image_gallery;
+        my $version = calc_img_version();
+        my $definition = get_required_var('DISTRI') . '-' . get_required_var('FLAVOR') . '-' . get_required_var('VERSION');
+        $definition = get_var("PUBLIC_CLOUD_AZURE_IMAGE_DEFINITION", uc($definition));
+        # asdf
+        $json = script_output("az sig image-version show --resource-group '$resource_group' --gallery-name '$gallery' " .
+              "--gallery-image-definition '$definition' --gallery-image-version '$version'", timeout => 60 * 30);
+        record_info('IMGV INFO', $json);
```

Comment:
this lines : 
```
record_info('IMGV INFO', $json);
        eval {
            $image = decode_azure_json($json)->{name};
        };
        record_info('IMGV NOT-FOUND', "Cannot find image-version $name. Need to upload it.\n$@") if ($@);
```
are fully identical in if/else hence can be moved out from if/else

Reply:
I don't understand, can you please help me out?

Reply:
lines 99,100,101 within `if ($arch eq 'Arm64')`  and lines 104,105,106 within `else` branch are identical so no matter what arch is we will do this , so there is no point to keep the duplicated inside each branch we can just call this AFTER `if/esle` : 

```
} else {
        $json = script_output("az image show --resource-group " . $self->resource_group . " --name $name", 60, proceed_on_failure => 1);
}

record_info('IMG INFO', $json);
eval { $image = decode_azure_json($json)->{name}; };
record_info('IMG NOT-FOUND', "Cannot find image $name. Need to upload it.\n$@") if ($@);
```